packages/apify/src/actor.ts

```diff
@@ -447,16 +447,16 @@ export class Actor<Data extends Dictionary = Dictionary> {
             return;
         }
 
-        // Waiting for all the listeners to finish, as `.metamorph()` kills the container.
+        // Waiting for all the listeners to finish, as `.reboot()` kills the container.
         await Promise.all([
             // `persistState` for individual RequestLists, RequestQueue... instances to be persisted
             ...this.config.getEventManager().listeners(EventType.PERSIST_STATE).map((x) => x()),
             // `migrating` to pause Apify crawlers
             ...this.config.getEventManager().listeners(EventType.MIGRATING).map((x) => x()),
         ]);
 
-        const actorId = this.config.get('actorId')!;
-        await this.metamorph(actorId);
+        const runId = this.config.get('actorRunId')!;
+        await this.apifyClient.run(runId).reboot();
```

Comment:
We should sleep here after the API call, like we do with metamorph, so that the Actor waits for the reboot to happen (otherwise it could finish before the reboot happens).

Reply:
you're right, missed that, thanks